### DISCUSSION TITLE
Stage name in the Request's URI

### DISCUSSION THREAD

**Discussion by seanpianka** [aws/aws-lambda-rust-runtime#463] [Q&A] [answered]
Hello! 👋 My problem is a bit specific, but I am upgrading from a pinned revision from Nov 2021 ([dcc33ea9e3fba5750f582ff5307a26ee320d70ed](https://github.com/awslabs/aws-lambda-rust-runtime/commit/dcc33ea9e3fba5750f582ff5307a26ee320d70ed)). During this upgrade, I noticed that the URI in the `Request` has begun to include the name of the stage. 

Could anyone point me to when this change was introduced? In the process of my upgrading, I will need to handle the stage name being a part of the URI.

Thanks!

Edit: I believe this commit is the culprit: https://github.com/awslabs/aws-lambda-rust-runtime/commit/3d5565df8590b0024d04aa5dbecd081e0d335972#diff-8d6dff89da49a82ad3db479309029ace94da82f17d113341a28d418f45ebbd36

**Comment by calavera** [comment, marked as answer]
Yes, that was a change to be more closely compatible with Api Gateway. Here's an example on how you can get the path without the stage part: https://github.com/awslabs/aws-lambda-rust-runtime/blob/main/lambda-http/examples/hello-raw-http-path.rs

**Reply by seanpianka** [reply]
Thanks for the quick response. Is there a trait I need to import to use this? I have `lambda_http::RequestExt` imported, but I still get `error[E0599]: no method named raw_http_path found for struct hyper::Request in the current scope`. I'm using 0.5.

**Reply by calavera** [reply]
Oh, I think we didn't make that available in a release yet 🤦 

You can configure your lambda to fetch the code from GitHub, it might take us a day or two to coordinate a release. This is the configuration:

```
[dependencies]
lambda_http = { git = "https://github.com/awslabs/aws-lambda-rust-runtime" }
```

**Reply by seanpianka** [reply]
No worries, I can wait until the next release! I implemented this functionality manually by checking for the stage in the request context, and splitting the URI by that stage name. 

It will work for now, until the next release 🙂 Thanks again!
